python-squid/src/main/java/org/sonar/python/semantic/SymbolTableBuilder.java

```diff
@@ -96,6 +97,9 @@ public void visitFileInput(FileInput fileInput) {
         scope.symbols.forEach(classDef::addClassField);
         scope.instanceAttributesByName.values().forEach(classDef::addInstanceField);
       });
+    scopesByRootTree.values().stream()
```

Comment:
not related to your PR : I am just wondering if we could just avoid 3 iterations over values and have only one with a plain old for loop. If you're up to it, I think it is worth a commit of refactoring at the end of your PR.